### PR TITLE
[New Hack] 055-ChaosStudio4AKS

### PR DESCRIPTION
**Pull request by JerryMSFT**
PR # 1 for ChaosStudio Hack

### CODE REVIEW

**Comment by JerryMSFT** [comment]
removed 55-ChaosStudio file (dunno where this came from)

File: 055-ChaosStudio4AKS/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+## Learning Objectives
+This “What the Hack” WTH is designed to introduce you to Azure Chaos Studios (Preview) and guide you through a series of hands-on challenges to accomplish
+the following:
+  
+* Leverage the Azure Chaos Studio to inject failure into an application/workload
+* Provide hands-on understanding of Chaos Engineering 
+* Understand how resilientcy can be achieved with Azure 
+
+In this WTH, you are the system owner of the Contoso Pizzeria Application (or you may bring your own application). Super Bowl Sunday is Contoso Pizza's busiest time of the year, the pizzeria
```
Comment (Whowong) [inline 055-ChaosStudio4AKS/README.md:19]: Do we want partners/customers to bring their own app?  Based on the challenges I wonder if you will encounter the same experience of the hack if their app has some resiliency built in already.

File: 055-ChaosStudio4AKS/Student/Challenge-01.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+During the experiment, were you able to order a pizza or perform your appplication's functionality? If not, what could you do to make your application resilient at the POD layer?  
+
+
+## Success Criteria
+
+- Verify Chaos Mesh is running on the Cluster
+- Verify Pod Chaos restarted the application's AKS pod
+- Observe any failure in the the application
+- Was your application available? 
+- If not how can you make it available during such an outtage?
```
Comment (Whowong) [inline 055-ChaosStudio4AKS/Student/Challenge-01.md]: Based on the solution for challenge one, I think we should make the description/success criteria more clear that the students need to implement a change to make the cluster resiliant at the pod layer and re-run the experiment.

File: 055-ChaosStudio4AKS/Student/Challenge-03.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+During the experiment, were you able to order a pizza? If not, what could you do to make your application more resilient
+
+
+## Success Criteria
+
+- Verify the experiment is running
+- Observe any failure(s)
+- Discuss with your coach how your application is (or was made) resilient
+- Verify all application traffic is routing to the surviving region
```
Comment (Whowong) [inline 055-ChaosStudio4AKS/Student/Challenge-03.md:41]: Lets consider rewording it without giving the answer to how they could make the application more resilient.  Maybe "Verify application is available after WestUS region is offline"

File: 055-ChaosStudio4AKS/Student/Challenge-04.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+Run your experiments in Dev/Test, do not run in Prod.
+
+
+## Tips
+1. You want your application to be available (healthy state) during failure.
+2. What kinds of faults and remediations come to mind from the previous challenges? 
+
+## Success Criteria
+
+- Chaos Studio injects fault(s) into your application.
```
Comment (Whowong) [inline 055-ChaosStudio4AKS/Student/Challenge-04.md:27]: We should update the success criteria here to include some verbiage where it includes implementing this in a CI/CD pipeline.

Also what is the expectation here for the faults?  Do they just implement the fault in the pipeline?  Or are some type of tests also in the pipeline to verify the app is healthy?

File: 055-ChaosStudio4AKS/Coach/README.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+**NOTE:** If you are a Hackathon participant, this is the answer guide. Don't cheat yourself by looking at these during the hack! Go learn something. :)
+
+## Coach's Guides
+
+- Challenge 00: **[Prerequisites - Ready, Set, GO!](./Solution-00.md)**
+	 - Prepare your workstation to work with Azure.
+- Challenge 01: **[Is your Application ready for the Super Bowl](./Solution-01.md)**
+	 - How does your application handle failure during large scale events?
+- Challenge 02: **[My AZ burned down, now what?](./Solution-02.md)**
```
Comment (Whowong) [inline 055-ChaosStudio4AKS/Coach/README.md:17]: Names of the solutions should match.  Please make sure to align with challenge titles as well.

**Review by Whowong — CHANGES_REQUESTED** [review]

File: 055-ChaosStudio4AKS/README.md (hunk 1 above)
Comment (JerryMSFT) [reply 055-ChaosStudio4AKS/README.md:19]: I think we had discussed this with Peter last year and he recommended we have an option to bring their own app

File: 055-ChaosStudio4AKS/Coach/README.md (hunk 5 above)
Comment (JerryMSFT) [reply 055-ChaosStudio4AKS/Coach/README.md:17]: I think I got them all, as they are little ones and my eyes are tired of this hack, I may have missed some

File: 055-ChaosStudio4AKS/Student/Challenge-03.md (hunk 3 above)
Comment (JerryMSFT) [reply 055-ChaosStudio4AKS/Student/Challenge-03.md:41]: changed to "Verify application is available after WestUS region is offline"

File: 055-ChaosStudio4AKS/Student/Challenge-04.md (hunk 4 above)
Comment (JerryMSFT) [reply 055-ChaosStudio4AKS/Student/Challenge-04.md:27]: Maybe you and I should work on this one. I have instructions for a simple VM test versus rigging up the Pizza App in a CI/CD pipeline

**Comment by JerryMSFT** [comment]
Please review changes

File: 055-ChaosStudio4AKS/README.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+## Challenges
+* Challenge 00: **[Ready Set Go](Student/Challenge-00.md)**
+	 - Deploy the multi-region Kubernetes pizzeria application,
+* Challenge 01: **[Is your application ready for the Super Bowl?](Student/Challenge-01.md)**
+	 - How does your application handle failure during large scale events?
+* Challenge 02: **[My AZ burned down, now what?](Student/Challenge-02.md)**
+	 - Can your application survive an Azure outage of 1 or more Availability Zones?
+* Challenge 03: **[Godzilla takes out an Azure region!](Student/Challenge-03.md)**
+	 - Can your application survive a region failure? 
+* Challenge 04: **[Injecting Chaos into your pipeline](Student/Challenge-04.md)**
```
[the remark this answers is not in the document]
Comment (JerryMSFT) [reply 055-ChaosStudio4AKS/README.md]: fixed

File: 055-ChaosStudio4AKS/README.md (hunk 1 above)
Comment (Whowong) [reply 055-ChaosStudio4AKS/README.md:19]: @jrzyshr - Tagging to comment.

**Review by Whowong — APPROVED** [review]

**Review by perktime — APPROVED** [review]

**Pull request merged**